pkg/apis/app/v1alpha1/kogitobuild_types.go

```diff
@@ -198,6 +203,99 @@ type KogitoBuildConditions struct {
 	Message string `json:"message,omitempty"`
 }
 
+// GitSource Git coordinates to locate the source code to build.
+// +k8s:openapi-gen=true
+// +operator-sdk:gen-csv:customresourcedefinitions.displayName="Kogito Git Source"
+type GitSource struct {
+	// Git URI for the s2i source.
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Git URI"
+	URI string `json:"uri"`
+	// Branch to use in the Git repository.
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Git Reference"
+	Reference string `json:"reference,omitempty"`
+	// Context/subdirectory where the code is located, relative to the repo root.
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Git Context"
+	ContextDir string `json:"contextDir,omitempty"`
+}
+
+// Builds ...
+// +k8s:openapi-gen=true
+type Builds struct {
+	// Builds are being created.
+	// +listType=set
+	New []string `json:"new,omitempty"`
+	// Builds are about to start running.
+	// +listType=set
+	Pending []string `json:"pending,omitempty"`
+	// Builds are running.
+	// +listType=set
+	Running []string `json:"running,omitempty"`
+	// Builds have executed and succeeded.
+	// +listType=set
+	Complete []string `json:"complete,omitempty"`
+	// Builds have executed and failed.
+	// +listType=set
+	Failed []string `json:"failed,omitempty"`
+	// Builds have been prevented from executing by an error.
+	// +listType=set
+	Error []string `json:"error,omitempty"`
+	// Builds have been stopped from executing.
+	// +listType=set
+	Cancelled []string `json:"cancelled,omitempty"`
+}
+
+// RuntimeType - type of condition.
+type RuntimeType string
+
+const (
+	// QuarkusRuntimeType Runtime for quarkus application
+	QuarkusRuntimeType RuntimeType = "quarkus"
+	// SpringbootRuntimeType Runtime for springboot application
+	SpringbootRuntimeType RuntimeType = "springboot"
+)
+
+// Artifact contains override information for building the Maven artifact.
+// + optional
+// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Final Artifact"
+type Artifact struct {
+
+	//Indicates the unique identifier of the organization or group that created the project.
+	// + optional
+	GroupID string `json:"groupId,omitempty"`
+
+	//Indicates the unique base name of the primary artifact being generated.
+	// + optional
+	ArtifactID string `json:"artifactId,omitempty"`
+
+	//Indicates the version of the artifact generated by the project.
+	// + optional
+	Version string `json:"version,omitempty"`
+}
+
+// WebhookType literal type to distinguish between different types of webhooks.
+type WebhookType string
```

Comment:
```suggestion
type WebHookType string
```
My IDE always annoys me because of this typo. I was waiting for this refactor to rename it. Could you? :D

Reply:
done